tests/static_multimap/pair_function_test.cu

```diff
@@ -124,11 +140,11 @@ TEMPLATE_TEST_CASE_SIG(
                           cuco::linear_probing<1, cuco::detail::MurmurHash3_32<Key>>>
       map{
         num_pairs * 2, cuco::sentinel::empty_key<Key>{-1}, cuco::sentinel::empty_value<Value>{-1}};
-    test_pair_functions<Key, Value>(map, d_pairs.begin(), num_pairs);
+    test_pair_functions<Key, Value>(map, d_pairs.data().get(), num_pairs);
```

Comment:
Why was this change needed?

Reply:
This is to get rid of the dereferenc error at https://github.com/PointKernel/cuCollections/blob/add-multimap-pair-contains/include/cuco/detail/static_multimap/device_view_impl.inl#L660
```bash
/home/yunsongw/Work/cuCollections/include/cuco/detail/static_multimap/device_view_impl.inl(660): error: class "thrust::device_reference<cuco::pair<int64_t, int64_t>>" has no member "first"
          detected during:
            instantiation of "std::enable_if_t<<expression>, __nv_bool> cuco::static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_impl::contains<is_pair_contains,uses_vector_load,ProbeT,Equal>(const cooperative_groups::__v1::thread_block_tile<ProbeSequence::cg_size, void> &, const ProbeT &, Equal) const noexcept [with Key=int64_t, Value=int64_t, Scope=cuda::std::__4::__detail::thread_scope_device, Allocator=cuco::cuda_allocator<char>, ProbeSequence=cuco::double_hashing<8U, cuco::detail::MurmurHash3_32<int64_t>, cuco::detail::MurmurHash3_32<int64_t>>, is_pair_contains=true, uses_vector_load=false, ProbeT=thrust::device_reference<cuco::pair<int64_t, int64_t>>, Equal=pair_equal<int64_t, int64_t>]" 
```

Reply:
I suspected it was something like that. 

cuco functions need to be able to work on Thrust iterators directly and we can't assume inputs will always be raw pointers like `data().get()`. 

I'll need to look further into the code here, but I suspect we need to do some coaxing of the type into the shape we expect.

Reply:
The iterator is dereferenced [here](https://github.com/PointKernel/cuCollections/blob/add-multimap-pair-contains/include/cuco/detail/static_multimap/kernels.cuh#L186) by using `*`.

Reply:
As we discussed offline, a short term fix is we can replace that line with:
```
std::iterator_traits<InputIt>::value_type element = *(first + idx);
```

Reply:
Fixed.